src/main/java/com/hivemq/mqtt/services/PublishPollServiceImpl.java

```diff
@@ -181,12 +181,16 @@ public void onSuccess(final ImmutableList<PUBLISH> publishes) {
                 for (int i = usedIds; i < messageIds.length(); i++) {
                     messageIDPool.returnId(messageIds.get(i));
                 }
-
+                final List<PublishWithFuture> publishesToSend = new ArrayList<>();
```

Comment:
```suggestion
                final List<PublishWithFuture> publishesToSend = new ArrayList<>(publishes.size());
```
or directly use an array for less garbage

Reply:
`sendOutPublishes()` takes a list as argument. Therfore, we would need to convert it back to a list anyway and I think a list is the way to go.

Reply:
Or we would need to change `sendPublishes()` to accept an array.